core/partitioning/segmentedblock/SegmentedBlock.h

```diff
@@ -72,18 +73,25 @@ struct SegmentedBlock {
   bool contain_raw_value(torch::jit::Value* input) const {
     return old_to_new_.count(input);
   }
-  void register_inshapes(std::vector<ir::Input>& in_shapes) {
-    in_shapes_ = in_shapes;
+  void register_inshapes(std::vector<std::vector<int64_t>>& in_shapes, const ir::ShapeMode& shape_mode) {
+    if (shape_mode == ir::ShapeMode::kMIN) {
+      min_shapes_ = in_shapes;
+    } else if (shape_mode == ir::ShapeMode::kOPT) {
+      opt_shapes_ = in_shapes;
+    } else {
+      max_shapes_ = in_shapes;
+    }
   }
-  const std::vector<ir::Input>& in_shapes() const {
-    return in_shapes_;
+  const std::vector<std::vector<int64_t>> in_shapes() const {
```

Comment:
Maybe we should rename this to `in_opt_shapes`? Since we have min, opt and max shapes now, this `in_shapes` seems not clear

Reply:
Done.